.github/workflows/codeql-analysis.yml

```diff
@@ -30,3 +30,34 @@ jobs:
         with:
           language: ${{ matrix.language }}
           config-file: ./.github/codeql/config.yml
+
+  assign-failure-label:
+    if: github.event_name == 'pull_request' && failure()
+    needs: [analyze]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
```

Comment:
Does this need to checkout the repo?

Reply:
Yes

Reply:
Well let me check to see. In the docs, it says to use it but maybe it doesn't.

Reply:
@ericboehs Yes, this is needed